data_steward/tools/recreate_person.py

```diff
@@ -7,20 +7,31 @@
 
 LOGGER = logging.getLogger(__name__)
 
-NEW_PERSON_QUERY = """
-SELECT
-    p.* EXCEPT (state_of_residence_concept_id, state_of_residence_source_value),
-    ext.state_of_residence_concept_id,
-    ext.state_of_residence_source_value
+ADD_COLUMNS_QUERY = JINJA_ENV.from_string("""
+ALTER TABLE `{{person.project}}.{{person.dataset_id}}.{{person.table_id}}`
+ADD COLUMN IF NOT EXISTS state_of_residence_concept_id INT64 
+    OPTIONS(description="[All of Us OMOP extension] A foreign key for the state of residence"),
+ADD COLUMN IF NOT EXISTS state_of_residence_source_value STRING 
+    OPTIONS(description="[All of Us OMOP extension] The source code for the state of residence")
+""")
+
+UPDATE_PERSON_QUERY = JINJA_ENV.from_string("""
+UPDATE `{{person.project}}.{{person.dataset_id}}.{{person.table_id}}` p
```

Comment:
Yes, our primary motivation for using DDL was the time it takes (also depends on the query, but for most cleaning rules it works well). However, because we use a `WRITE_TRUNCATE` we lose the schema. DML is advantageous in preserving the schema and since the query affects only 2 columns and BQ is architecture columnar, this performs relatively quickly. I tried to `WRITE_TRUNCATE` to a specific partition as we did earlier to preserve the schema but it looks like the Python BQ client doesn't have that capability yet.

Reply:
That's right, that was our main concern was the schema being wiped out. Thanks for explaining this!